lib/sass_spec/stats/todo.rb

```diff
@@ -0,0 +1,240 @@
+# encoding: utf-8
+
+require 'pathname'
+require 'command_line_reporter'
+
+class SassSpec::Stats::Todo
+  include CommandLineReporter
+
+  def self.parse(args)
+    options = {}
+    parser = OptionParser.new
+    parser.banner = "Usage: ./sass-spec.rb todo [filter]\n\nThis sub command prints stats on todos.\n\n"
+
+    parser.on("--impl IMPLEMENTATION",
+              "Find todos for implementation.") do |impl|
+      if !["dart-sass", "libsass"].include? impl
+        warn "Implementation must be one of [dart-sass, libsass].\n\n"
+        warn parser.help()
+        return nil
+      end
+      options[:impl] = impl
+    end
+
+    parser.on("--issue GITHUB_ISSUE",
+              "Find todos corresponding to a Github issue in the form `dart-sass#123`.") do |issue|
+      ["dart-sass", "libsass"].each do |impl|
+        regex = Regexp.new("^#{impl}#\\d+$")
+        if issue.match(regex)
+          options[:issue] = issue
+        end
+      end
+
+      if !options[:issue]
+        warn "Invalid issue.\n\n"
+        warn parser.help()
+        return nil
+      end
+    end
+
+    parser.on("--missing-issue", "Find todos that are missing a Github issue.") do
+      options[:missing_issue] = true
+    end
+
+    parser.on("-h", "--help", "Print this help message.") do
+      puts parser.help()
+      return nil
+    end
+
+    parser.parse!(args)
+    new(options)
+  end
+
+  def initialize(options)
+    @options = options
+  end
+
+  def stats
+    files = Dir.glob(File.join(SassSpec::SPEC_DIR, '**/*')).uniq
+
+    if @options[:impl]
+      todos_by_implementation(files, @options[:impl])
+    elsif @options[:issue]
+      todos_by_issue(files, @options[:issue])
+    elsif @options[:missing_issue]
+      todos_missing_issue(files)
+    else
+      todos(files)
+    end
+  end
+
+  def todos_by_implementation(files, implementation)
+    regexes = []
+    if implementation == "dart-sass"
+      dart_sass = Regexp.new(':todo:(?:\n- libsass)*?(?:\n- sass\/libsass)*?\n- sass\/dart-sass#\d+$', Regexp::MULTILINE)
```

Comment:
Why not search the parsed YAML contents of these files rather than use regexps? It seems like that would be more robust and probably easier to read as well.

Reply:
Great idea, thanks